crates/cranelift/src/func_environ.rs

```diff
@@ -284,14 +285,13 @@ impl<'module_environment> FuncEnvironment<'module_environment> {
         // function. This is possible since we know that the pointer never
         // changes for the lifetime of the function.
         let pointer_type = self.pointer_type();
-        builder.declare_var(self.vmruntime_limits_ptr, pointer_type);
         let vmctx = self.vmctx(builder.func);
         let base = builder.ins().global_value(pointer_type, vmctx);
         let offset = i32::try_from(self.offsets.vmctx_runtime_limits()).unwrap();
         let interrupt_ptr = builder
             .ins()
             .load(pointer_type, ir::MemFlags::trusted(), base, offset);
-        builder.def_var(self.vmruntime_limits_ptr, interrupt_ptr);
+        self.vmruntime_limits_ptr = interrupt_ptr;
```

Comment:
Can we debug assert that `self.vmruntime_limits_ptr` is the reserved value before we initialize it, to make sure that we aren't defining it multiple times? And/or use an `Option` or `PackedOption` for the field?